src/views/HomeView.vue

```diff
@@ -0,0 +1,35 @@
+<template>
+  <div>
+    <header-component>
+      <search-bar/>
+    </header-component>
+    <result-filter/>
+    <movie-search-result/>
+  </div>
+</template>
+
+<script>
+import MovieSearchResult from "@/components/MovieGrid";
+import ResultFilter from "@/components/ResultFilter";
+import HeaderComponent from "@/components/HeaderComponent";
+import SearchBar from "@/components/SearchBar";
+
+export default {
+  name: "HomeView",
+  components: {
+    SearchBar,
+    MovieSearchResult,
+    ResultFilter,
+    HeaderComponent
+  },
+  created() {
+    if (this.$store.state.movies.length === 0) {
```

Comment:
This could be placed in the component state and used to conditionally show a loader while the request is in flight.